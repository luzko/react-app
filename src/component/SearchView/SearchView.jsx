import React, {useEffect} from "react";
import Header from "../Header";
import Main from "../Main";

import {useLocation, useParams} from 'react-router-dom';

const SearchView = (props) => {
  let location = useLocation();
  let {title} = useParams();

  useEffect(() => {
    if (location.pathname !== '/') {
      props.setSearch(title)
    }
    props.processing();
    props.fetchMovies();
  }, []);

  return (
      <>
        <Header
            title={title}
            createMovie={props.createMovie}
            setSearch={props.setSearch}
            processing={props.processing}
            fetchMovies={props.fetchMovies}
        />
        <Main
            isLoad={props.isLoad}
            error={props.error}
            movies={props.movies}
            deleteMovie={props.deleteMovie}
            updateMovie={props.updateMovie}
        />
      </>
  );
};

export default SearchView;
