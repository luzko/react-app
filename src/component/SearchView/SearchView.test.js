import React from 'react';
import renderer from 'react-test-renderer';
import SearchView from './SearchView';

import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import {MemoryRouter} from 'react-router';

const mockStore = configureStore();

const initialState = {
  processing: false,
  error: null,
  movie: null,
  movies: [
    {
      'id': 1,
      'title': 'mock movie 1',
      'tagline': 'mock tagline 1',
      'vote_average': 9.9,
      'vote_count': 0,
      'release_date': '2021-04-18',
      'poster_path': 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      'overview': 'mock overview',
      'budget': 1000,
      'revenue': 2000,
      'genres': ['Drama', 'Romance'],
      'runtime': 90
    },
    {
      'id': 2,
      'title': 'mock movie 2',
      'tagline': 'mock tagline 2',
      'vote_average': 9.9,
      'vote_count': 0,
      'release_date': '2021-01-01',
      'poster_path': 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
      'overview': 'mock overview 2',
      'budget': 1000,
      'revenue': 2000,
      'genres': ['Drama', 'Romance'],
      'runtime': 100
    },
  ],
  filterSort: {
    sortBy: 'vote_average',
    sortOrder: 'desc',
    filter: '',
    search: '',
    searchBy: 'title'
  }
}

const store = mockStore(initialState);

test('snapshot test for search view', () => {
  const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <SearchView/>
        </Provider>
      </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
