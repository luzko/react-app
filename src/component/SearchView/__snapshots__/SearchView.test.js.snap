// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for search view 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="logoAndButton"
    >
      <img
        alt="logo"
        className="logo"
        src="logo.svg"
      />
      <button
        className="addButton"
        onClick={[Function]}
      >
        + ADD MOVIE
      </button>
      <div
        className="modal "
        onClick={[Function]}
      >
        <div
          className="content"
          onClick={[Function]}
        >
          <button
            className="close"
            onClick={[Function]}
          >
            ✕
          </button>
          <p
            className="title"
          >
            ADD MOVIE
          </p>
          <form
            onSubmit={[Function]}
          >
            <div
              className="field"
            >
              <label>
                TITLE
              </label>
              <input
                className="input "
                id="title"
                name="title"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="field"
            >
              <label>
                TAGLINE
              </label>
              <input
                className="input "
                id="tagline"
                name="tagline"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="field"
            >
              <label>
                RELEASE DATE
              </label>
              <input
                className="input "
                id="release"
                name="release"
                onBlur={[Function]}
                onChange={[Function]}
                type="date"
                value=""
              />
            </div>
            <div
              className="field"
            >
              <label>
                POSTER
              </label>
              <input
                className="input "
                id="poster"
                name="poster"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="field"
            >
              <label>
                RATING
              </label>
              <input
                className="input "
                id="rating"
                name="rating"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="field"
            >
              <label>
                GENRES
              </label>
              <input
                className="input "
                disabled={true}
                id="genre"
                name="genre"
                onBlur={[Function]}
                type="text"
                value={Array []}
              />
              <div
                className="arrow"
                onClick={[Function]}
              >
                ▼
              </div>
              <div
                className="list hidden"
              >
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Fantasy"
                  />
                  <label>
                    Fantasy
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Drama"
                  />
                  <label>
                    Drama
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Action"
                  />
                  <label>
                    Action
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Comedy"
                  />
                  <label>
                    Comedy
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Crime"
                  />
                  <label>
                    Crime
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Adventure"
                  />
                  <label>
                    Adventure
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Thriller"
                  />
                  <label>
                    Thriller
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Family"
                  />
                  <label>
                    Family
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Documentary"
                  />
                  <label>
                    Documentary
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Horror"
                  />
                  <label>
                    Horror
                  </label>
                </li>
              </div>
            </div>
            <div
              className="field"
            >
              <label>
                RUNTIME
              </label>
              <input
                className="input "
                id="runtime"
                name="runtime"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="field"
            >
              <label>
                OVERVIEW
              </label>
              <textarea
                className="input "
                id="overview"
                name="overview"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
            <div
              className="buttons"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                RESET
              </button>
              <button
                className="button"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      className="headerText"
    >
      FIND YOUR MOVIE
    </div>
    <div
      className="search"
    >
      <input
        className="searchInput"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="What do you want to watch?"
        type="text"
        value=""
      />
      <button
        className="searchButton"
        onClick={[Function]}
        title="search"
      >
        SEARCH
      </button>
    </div>
  </header>,
  <main
    className="main"
  >
    <div
      className="filterSort"
    >
      <div
        className="genreList"
      >
        <button
          className="option active"
          onClick={[Function]}
        >
          ALL
        </button>
        <button
          className="option "
          onClick={[Function]}
        >
          Documentary
        </button>
        <button
          className="option "
          onClick={[Function]}
        >
          Comedy
        </button>
        <button
          className="option "
          onClick={[Function]}
        >
          Horror
        </button>
        <button
          className="option "
          onClick={[Function]}
        >
          Crime
        </button>
      </div>
      <div>
        <span
          className="label"
        >
          SORT BY:
        </span>
        <select
          className="dropdown"
          defaultValue={null}
          onChange={[Function]}
        >
          <option
            value="release_date"
          >
            RELEASE DATE
          </option>
          <option
            value="title"
          >
            NAME
          </option>
          <option
            value="vote_average"
          >
            RATING
          </option>
          <option
            value="runtime"
          >
            RUNTIME
          </option>
        </select>
        <div
          className="order"
          onClick={[Function]}
        >
          ▼
        </div>
      </div>
    </div>
    <div
      className="count"
    >
      2
       movies found
    </div>
    <div
      className="movieList"
    >
      <div
        className="card"
      >
        <div
          className="container"
        >
          <img
            alt="mock movie 1"
            className="image"
            onClick={[Function]}
            src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
          />
          <button
            className="button"
            onClick={[Function]}
          >
            ...
          </button>
        </div>
        <div
          className="name"
        >
          <div>
            mock movie 1
          </div>
          <div>
            2021
          </div>
        </div>
        <div
          className="genres"
        >
          Drama, Romance
        </div>
        <div
          className="options"
        >
          <div
            className="movieOptions hidden"
            onMouseLeave={[Function]}
          >
            <div
              className="options"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                EDIT
              </button>
              <button
                className="button"
                onClick={[Function]}
              >
                DELETE
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal "
        onClick={[Function]}
      >
        <div
          className="content"
          onClick={[Function]}
        >
          <button
            className="close"
            onClick={[Function]}
          >
            ✕
          </button>
          <p
            className="title"
          >
            Delete Movie
          </p>
          <div
            className="delete"
          >
            <p>
              Are you sure you want to delete this movie?
            </p>
            <div
              className="buttons"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal "
        onClick={[Function]}
      >
        <div
          className="content"
          onClick={[Function]}
        >
          <button
            className="close"
            onClick={[Function]}
          >
            ✕
          </button>
          <p
            className="title"
          >
            EDIT MOVIE
          </p>
          <form
            onSubmit={[Function]}
          >
            <div
              className="field"
            >
              <label>
                TITLE
              </label>
              <input
                className="input "
                id="title"
                name="title"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="mock movie 1"
              />
            </div>
            <div
              className="field"
            >
              <label>
                TAGLINE
              </label>
              <input
                className="input "
                id="tagline"
                name="tagline"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="mock tagline 1"
              />
            </div>
            <div
              className="field"
            >
              <label>
                RELEASE DATE
              </label>
              <input
                className="input "
                id="release"
                name="release"
                onBlur={[Function]}
                onChange={[Function]}
                type="date"
                value="2021-04-18"
              />
            </div>
            <div
              className="field"
            >
              <label>
                POSTER
              </label>
              <input
                className="input "
                id="poster"
                name="poster"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
              />
            </div>
            <div
              className="field"
            >
              <label>
                RATING
              </label>
              <input
                className="input "
                id="rating"
                name="rating"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value={9.9}
              />
            </div>
            <div
              className="field"
            >
              <label>
                GENRES
              </label>
              <input
                className="input "
                disabled={true}
                id="genre"
                name="genre"
                onBlur={[Function]}
                type="text"
                value={
                  Array [
                    "Drama",
                    "Romance",
                  ]
                }
              />
              <div
                className="arrow"
                onClick={[Function]}
              >
                ▼
              </div>
              <div
                className="list hidden"
              >
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Fantasy"
                  />
                  <label>
                    Fantasy
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={true}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Drama"
                  />
                  <label>
                    Drama
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Action"
                  />
                  <label>
                    Action
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Comedy"
                  />
                  <label>
                    Comedy
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Crime"
                  />
                  <label>
                    Crime
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Adventure"
                  />
                  <label>
                    Adventure
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Thriller"
                  />
                  <label>
                    Thriller
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Family"
                  />
                  <label>
                    Family
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Documentary"
                  />
                  <label>
                    Documentary
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Horror"
                  />
                  <label>
                    Horror
                  </label>
                </li>
              </div>
            </div>
            <div
              className="field"
            >
              <label>
                RUNTIME
              </label>
              <input
                className="input "
                id="runtime"
                name="runtime"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value={90}
              />
            </div>
            <div
              className="field"
            >
              <label>
                OVERVIEW
              </label>
              <textarea
                className="input "
                id="overview"
                name="overview"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="mock overview"
              />
            </div>
            <div
              className="buttons"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                RESET
              </button>
              <button
                className="button"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        className="card"
      >
        <div
          className="container"
        >
          <img
            alt="mock movie 2"
            className="image"
            onClick={[Function]}
            src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
          />
          <button
            className="button"
            onClick={[Function]}
          >
            ...
          </button>
        </div>
        <div
          className="name"
        >
          <div>
            mock movie 2
          </div>
          <div>
            2021
          </div>
        </div>
        <div
          className="genres"
        >
          Drama, Romance
        </div>
        <div
          className="options"
        >
          <div
            className="movieOptions hidden"
            onMouseLeave={[Function]}
          >
            <div
              className="options"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                EDIT
              </button>
              <button
                className="button"
                onClick={[Function]}
              >
                DELETE
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal "
        onClick={[Function]}
      >
        <div
          className="content"
          onClick={[Function]}
        >
          <button
            className="close"
            onClick={[Function]}
          >
            ✕
          </button>
          <p
            className="title"
          >
            Delete Movie
          </p>
          <div
            className="delete"
          >
            <p>
              Are you sure you want to delete this movie?
            </p>
            <div
              className="buttons"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal "
        onClick={[Function]}
      >
        <div
          className="content"
          onClick={[Function]}
        >
          <button
            className="close"
            onClick={[Function]}
          >
            ✕
          </button>
          <p
            className="title"
          >
            EDIT MOVIE
          </p>
          <form
            onSubmit={[Function]}
          >
            <div
              className="field"
            >
              <label>
                TITLE
              </label>
              <input
                className="input "
                id="title"
                name="title"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="mock movie 2"
              />
            </div>
            <div
              className="field"
            >
              <label>
                TAGLINE
              </label>
              <input
                className="input "
                id="tagline"
                name="tagline"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="mock tagline 2"
              />
            </div>
            <div
              className="field"
            >
              <label>
                RELEASE DATE
              </label>
              <input
                className="input "
                id="release"
                name="release"
                onBlur={[Function]}
                onChange={[Function]}
                type="date"
                value="2021-01-01"
              />
            </div>
            <div
              className="field"
            >
              <label>
                POSTER
              </label>
              <input
                className="input "
                id="poster"
                name="poster"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
              />
            </div>
            <div
              className="field"
            >
              <label>
                RATING
              </label>
              <input
                className="input "
                id="rating"
                name="rating"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value={9.9}
              />
            </div>
            <div
              className="field"
            >
              <label>
                GENRES
              </label>
              <input
                className="input "
                disabled={true}
                id="genre"
                name="genre"
                onBlur={[Function]}
                type="text"
                value={
                  Array [
                    "Drama",
                    "Romance",
                  ]
                }
              />
              <div
                className="arrow"
                onClick={[Function]}
              >
                ▼
              </div>
              <div
                className="list hidden"
              >
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Fantasy"
                  />
                  <label>
                    Fantasy
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={true}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Drama"
                  />
                  <label>
                    Drama
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Action"
                  />
                  <label>
                    Action
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Comedy"
                  />
                  <label>
                    Comedy
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Crime"
                  />
                  <label>
                    Crime
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Adventure"
                  />
                  <label>
                    Adventure
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Thriller"
                  />
                  <label>
                    Thriller
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Family"
                  />
                  <label>
                    Family
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Documentary"
                  />
                  <label>
                    Documentary
                  </label>
                </li>
                <li
                  className="li"
                >
                  <input
                    checked={false}
                    name="genre"
                    onChange={[Function]}
                    type="checkbox"
                    value="Horror"
                  />
                  <label>
                    Horror
                  </label>
                </li>
              </div>
            </div>
            <div
              className="field"
            >
              <label>
                RUNTIME
              </label>
              <input
                className="input "
                id="runtime"
                name="runtime"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value={100}
              />
            </div>
            <div
              className="field"
            >
              <label>
                OVERVIEW
              </label>
              <textarea
                className="input "
                id="overview"
                name="overview"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="mock overview 2"
              />
            </div>
            <div
              className="buttons"
            >
              <button
                className="button"
                onClick={[Function]}
              >
                RESET
              </button>
              <button
                className="button"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </main>,
]
`;
